include/cppkafka/utils/buffered_producer.h

```diff
@@ -358,6 +373,18 @@ class CPPKAFKA_API BufferedProducer {
      * Simple helper to construct a builder object
      */
     Builder make_builder(std::string topic);
+    
+    /**
+     * Set the type of notification when RD_KAFKA_RESP_ERR__QUEUE_FULL is received.
+     *
+     * This will call the error callback for this producer. By default this is set to QueueFullNotification::None.
+     */
+    void set_queue_full_notification(QueueFullNotification notification);
```

Comment:
Maybe the getter for the notifiaction enum and the callback should be done in just one? You'll always call either both or none of them so I think that would make sense.

Reply:
You mean the setter of the notification type? I think there might be cases when the app wants to set the notifocation to some enum and then it might want to turn it off (None) for a period and re-enable it later. In that case it would have to set the callback each time...prob not desirable.

Reply:
Alright, fair enough. Merging this one, thanks!